src/mantine-styles/src/theme/types/MantineTheme.ts

```diff
@@ -98,9 +99,6 @@ interface ThemeComponent {
 }
 
 export type MantineThemeBase = Omit<MantineTheme, 'fn'>;
-export type MantineThemeOverride = DeepPartial<
-  Omit<MantineThemeBase, 'fn' | 'other' | 'components'>
-> & {
-  other?: MantineThemeOther;
-  components?: Record<string, ThemeComponent>;
-};
+
+export type MantineThemeOverride = DeepPartial<Omit<MantineThemeBase, 'other' | 'components'>> &
+  Partial<Pick<MantineThemeBase, 'other' | 'components'>>;
```

Comment:
better and safer to `Pick` directly from `MantineThemeBase`, otherwise the types could fall out of sync